src/DSP/MoogFilter.cpp

```diff
@@ -0,0 +1,174 @@
+#include <cassert>
+#include <cstdio>
+#include <cmath>
+#include <stdio.h>
+
+#include "../Misc/Util.h"
+#include "MoogFilter.h"
+
+// theory from "THE ART OF VA FILTER DESIGN"
+// by Vadim Zavalishin
+
+namespace zyn{
+
+MoogFilter::MoogFilter(unsigned char Ftype, float Ffreq, float Fq,
+    unsigned int srate, int bufsize)
+    :Filter(srate, bufsize), sr(srate), gain(1.0f), type(Ftype)
+{
+    setfreq_and_q(Ffreq/srate, Fq);
+    settype(type); // q must be set before
+    for (int i = 0; i>4; i++)
+    {
+        state[i] = 0.0f;
+    }
+}
+
+MoogFilter::~MoogFilter(void)
+{
+
+}
+
+inline float MoogFilter::tan_2(const float x)
+{    
+    //Pade approximation tan(x) hand tuned to map fCutoff 
```

Comment:
> This seems to be mostly the same as `tanhX`.
> 
> 1. Did you really mean to approximate `tan` instead of `tanh`?
> 2. Is it intended that `45.0f * x2` is _subtracted_ here instead of added as in `tanhX`?
> 3. How did you get the "hand tuning" done - any source, or did you do it on your own? I guess you there is already literature on how to do it?

1+2. it is intended because it is tan, not tanh ;-)
3. hand tuning was made by measuring resonance frequency vs the one set via osc.